### PR TITLE
Credit card payments (without Save card details, 3D Secure)

### PR DESCRIPTION
**Pull request by maassenbas**
CSE credit card payment 
Result: Authorised CC payment in Adyen CA

### CODE REVIEW

File: cartridges/LINK_adyen/cartridge/client/default/js/adyen-cse.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+$('button[value="submit-payment"]').on('click', function (e) {
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/client/default/js/adyen-cse.js]: Thought there was a better way of doing this ?

File: cartridges/LINK_adyen/cartridge/controllers/Adyen.js [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+//         PaRes: req.form.PaRes //req.httpParameterMap.get("PaRes").stringValue
+//     }
+//     req.session.privacyCache.set('adyenResponse', adyenResponse);
+//     res.render('adyenpaymentredirect', {
+//         AdyenResponse: adyenResponse,
+//         ContinueURL: URLUtils.https('Adyen-AuthorizeWithForm')
+//     });
+//     next();
+// });
+
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/controllers/Adyen.js]: please remove commented code

File: cartridges/LINK_adyen/cartridge/controllers/Adyen.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        MD: req.querystring.MD,
+        PaRes: req.form.PaRes //req.httpParameterMap.get("PaRes").stringValue
+    }
+    req.session.privacyCache.set('adyenResponse', adyenResponse);
+    Logger.getLogger("Adyen").error("authorizeWithForm res " + JSON.stringify(res) + "req: " +  JSON.stringify(adyenResponse));
+    // var	adyen3DVerification = require('int_adyen/cartridge/scripts/adyen3DVerification'), result,
+    //     order = session.custom.order,
+    //     paymentInstrument = session.custom.paymentInstrument,
+    //     adyenResponse  = session.custom.adyenResponse;
+
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/controllers/Adyen.js]: please remove commented code

File: cartridges/LINK_adyen/cartridge/controllers/Adyen.js [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    var	adyen3DVerification = require('int_adyen/cartridge/scripts/adyen3DVerification'), result;
+        //order = session.custom.order,
+        //paymentInstrument = session.custom.paymentInstrument;
+
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/controllers/Adyen.js]: please remove commented code

File: cartridges/LINK_adyen/cartridge/controllers/Adyen.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    Logger.getLogger("Adyen").error("authorize3DBefore " + adyenResponse);
+    Transaction.begin();
+    result = adyen3DVerification.verify({
+        Order: order,
+        //Amount: paymentInstrument.paymentTransaction.amount,
+        //PaymentInstrument: paymentInstrument,
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/controllers/Adyen.js]: please remove commented code

File: cartridges/LINK_adyen/cartridge/controllers/Adyen.js [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        CurrentSession: req.session,
+        CurrentRequest: req.request,
+        MD: adyenResponse.MD,
+        PaResponse: adyenResponse.PaRes
+    });
+
+    //TODOBAS return back to Place Order call
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/controllers/Adyen.js]: what is this ?

File: cartridges/LINK_adyen/cartridge/controllers/Adyen.js [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+    return next();
+    //return result;
+    // res.render('adyenpaymentredirect', {
+    //     // ContinueURL: IssuerURL,
+    //     // IssuerURL: IssuerURL,
+    //     // PaRequest: PaRequest,
+    //     // TermURL: TermURL,
+    //     // Order: OrderObj
+    //     ContinueURL: URLUtils.https('Adyen-AuthorizeWithForm')
+    // });
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/controllers/Adyen.js]: please remove commented code

File: cartridges/LINK_adyen/cartridge/templates/default/adyenform.isml [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+        </form>
+        <script language="javascript">
+            var loaded = false;
+            function load(){
+                document.getElementById('3dform').submit();
+                loaded = true;
+            }
+            load();
+        </script>
+    </div>
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/templates/default/adyenform.isml]: This is not in scope of this PR right ?

File: cartridges/LINK_adyen/cartridge/templates/adyenpaymentredirect.isml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+			loaded = true;
+		}
+	</script>
+</head>
+<body onload="load();">
+	<form action="${URLUtils.continueURL()}" method="POST" name="RedirectForm" target="_parent">
+		<input type="hidden" name="PaReq" value="${pdict.paRequest}">
+		<input type="hidden" name="MD" value="${pdict.md}">
+	</form>
+</body>
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/templates/adyenpaymentredirect.isml]: This is not in scope of this PR right ?

File: cartridges/LINK_adyen/cartridge/controllers/Adyen.js [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+    if(result.error) {
+        res.json({
+            error: true,
+            errorMessage: Resource.msg('error.technical', 'checkout', null)
+        });
+        return next();
+    }
+
+    // Delete this part, it is copied from Place Order
+    Logger.getLogger("Adyen").error("PlaceOrderAuthorise3Dafter2");
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/controllers/Adyen.js]: remove this code and comment

File: cartridges/LINK_adyen/cartridge/controllers/Adyen.js [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+    Transaction.commit();
+    COHelpers.sendConfirmationEmail(order, req.locale.id);
+
+    // res.json({
+    //     error: false,
+    //     orderID: order.orderNo,
+    //     orderToken: order.orderToken,
+    //     continueUrl: URLUtils.url('Order-Confirm').toString()
+    // });
+
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/controllers/Adyen.js]: please remove commented code

File: cartridges/LINK_adyen/cartridge/controllers/Adyen.js [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+        //Amount: paymentInstrument.paymentTransaction.amount,
+        //PaymentInstrument: paymentInstrument,
+        CurrentSession: req.session,
+        CurrentRequest: req.request,
+        MD: adyenResponse.MD,
+        PaResponse: adyenResponse.PaRes
+    });
+
+    //TODOBAS return back to Place Order call
+    Logger.getLogger("Adyen").error("authorize3DResult " + JSON.stringify(result));
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/controllers/Adyen.js]: why logging it as error ?

File: cartridges/LINK_adyen/cartridge/controllers/Adyen.js [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+var server = require('server');
+
+var Logger = require('dw/system/Logger');
+var URLUtils = require('dw/web/URLUtils');
+var Transaction = require('dw/system/Transaction');
+var COHelpers = require('*/cartridge/scripts/checkout/checkoutHelpers');
+var OrderMgr = require('dw/order/OrderMgr');
+
+server.get('Adyen3D', server.middleware.https, function (req, res, next) {
+     Logger.getLogger("Adyen").error("Adyen3DCall res " + JSON.stringify(res) + "req: " +  JSON.stringify(req));
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/controllers/Adyen.js]: why log it as error ?

File: cartridges/LINK_adyen/cartridge/controllers/Adyen.js [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+            errorMessage: Resource.msg('error.technical', 'checkout', null)
+        });
+        return next();
+    }
+
+    // Delete this part, it is copied from Place Order
+    Logger.getLogger("Adyen").error("PlaceOrderAuthorise3Dafter2");
+    // Places the order
+    var placeOrderResult = COHelpers.placeOrder(order);
+    Logger.getLogger("Adyen").error("placeOrderResult: " + JSON.stringify(placeOrderResult));
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/controllers/Adyen.js]: why logging it as error ?

File: cartridges/LINK_adyen/cartridge/scripts/checkout/adyenHelpers.js [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+
+/**
+ * saves payment instruemnt to customers wallet
+ * @param {Object} billingData - billing information entered by the user
+ * @param {dw.order.Basket} currentBasket - The current basket
+ * @param {dw.customer.Customer} customer - The current customer
+ * @returns {dw.customer.CustomerPaymentInstrument} newly stored payment Instrument
+ */
+function savePaymentInstrumentToWallet(billingData, currentBasket, customer) {
+    Logger.getLogger("Adyen").error("savePaymentInstrumentToWallet billingData: " + JSON.stringify(billingData) + "currentBasket: " + JSON.stringify(currentBasket));
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/scripts/checkout/adyenHelpers.js]: why log it as error ?

File: cartridges/LINK_adyen/cartridge/controllers/Adyen.js [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+//     next();
+// });
+
+server.post('AuthorizeWithForm', server.middleware.https, function (req, res, next) {
+    var adyenResponse = {
+        MD: req.querystring.MD,
+        PaRes: req.form.PaRes //req.httpParameterMap.get("PaRes").stringValue
+    }
+    req.session.privacyCache.set('adyenResponse', adyenResponse);
+    Logger.getLogger("Adyen").error("authorizeWithForm res " + JSON.stringify(res) + "req: " +  JSON.stringify(adyenResponse));
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/controllers/Adyen.js]: why log it as error ?

File: cartridges/LINK_adyen/cartridge/controllers/Adyen.js [hunk 17]
```diff
@@ -0,0 +1,131 @@
...
+    // var	adyen3DVerification = require('int_adyen/cartridge/scripts/adyen3DVerification'), result,
+    //     order = session.custom.order,
+    //     paymentInstrument = session.custom.paymentInstrument,
+    //     adyenResponse  = session.custom.adyenResponse;
+
+    var	adyen3DVerification = require('int_adyen/cartridge/scripts/adyen3DVerification'), result;
+        //order = session.custom.order,
+        //paymentInstrument = session.custom.paymentInstrument;
+
+    Logger.getLogger("Adyen").error("authorize3DBefore " + adyenResponse);
```
Comment (rikterbeek) [inline cartridges/LINK_adyen/cartridge/controllers/Adyen.js]: why log it as error ?

**Comment by rikterbeek** [comment]
@maassenbas can you remove the .DS_Store file. Is the .tern-project file needed ?

File: cartridges/LINK_adyen/cartridge/templates/default/checkout/billing/alternativePaymentMethodForm.isml [hunk 18]
```diff
@@ -0,0 +1,4 @@
+    <!--- payment method is Adyen --->
```
Comment (lancergr) [inline cartridges/LINK_adyen/cartridge/templates/default/checkout/billing/alternativePaymentMethodForm.isml]: remove?

File: cartridges/LINK_adyen/cartridge/templates/default/checkout/billing/paymentOptions/creditCardContent.isml [hunk 19]
```diff
@@ -5,7 +5,7 @@
             name="${pdict.forms.billingForm.paymentMethod.htmlName}"
             value="CREDIT_CARD"
     >
-
+    <!--<input type="hidden" class="form-control adyenEncryptedData" id="adyenEncryptedData" name="${pdict.forms.billingForm.paymentMethod.htmlName}" encoding="off"/>-->
```
Comment (lancergr) [inline cartridges/LINK_adyen/cartridge/templates/default/checkout/billing/paymentOptions/creditCardContent.isml]: remove?

File: cartridges/LINK_adyen/cartridge/scripts/hooks/payment/processor/adyen_credit.js [hunk 20]
```diff
@@ -0,0 +1,127 @@
+/**
+ *
+ */
```
Comment (Aleffio) [inline cartridges/LINK_adyen/cartridge/scripts/hooks/payment/processor/adyen_credit.js]: remove

**Pull request merged**